src/systemd/sd-json.h

```diff
@@ -0,0 +1,334 @@
+        SD_JSON_FORMAT_COLOR            = 1 << 3, /* insert ANSI color sequences */
+        SD_JSON_FORMAT_COLOR_AUTO       = 1 << 4, /* insert ANSI color sequences if colors_enabled() says so */
+        SD_JSON_FORMAT_SOURCE           = 1 << 5, /* prefix with source filename/line/column */
+        SD_JSON_FORMAT_SSE              = 1 << 6, /* prefix/suffix with W3C server-sent events */
+        SD_JSON_FORMAT_SEQ              = 1 << 7, /* prefix/suffix with RFC 7464 application/json-seq */
+        SD_JSON_FORMAT_FLUSH            = 1 << 8, /* call fflush() after dumping JSON */
+        SD_JSON_FORMAT_EMPTY_ARRAY      = 1 << 9, /* output "[]" for empty input */
+        SD_JSON_FORMAT_OFF              = 1 << 10, /* make json_variant_format() fail with -ENOEXEC */
+        SD_JSON_FORMAT_CENSOR_SENSITIVE = 1 << 11, /* Replace all sensitive elements with the string "<sensitive data>" */
+        _SD_ENUM_FORCE_S64(JSON_FORMAT_FLAGS)
+} sd_json_format_flags_t;
+
+int sd_json_variant_format(sd_json_variant *v, sd_json_format_flags_t flags, char **ret);
+int sd_json_variant_dump(sd_json_variant *v, sd_json_format_flags_t flags, FILE *f, const char *prefix);
+
+int sd_json_variant_filter(sd_json_variant **v, char **to_remove);
+
+int sd_json_variant_set_field(sd_json_variant **v, const char *field, sd_json_variant *value);
+int sd_json_variant_set_fieldb(sd_json_variant **v, const char *field, ...);
+int sd_json_variant_set_field_string(sd_json_variant **v, const char *field, const char *value);
+int sd_json_variant_set_field_integer(sd_json_variant **v, const char *field, int64_t value);
+int sd_json_variant_set_field_unsigned(sd_json_variant **v, const char *field, uint64_t value);
+int sd_json_variant_set_field_boolean(sd_json_variant **v, const char *field, int b);
+int sd_json_variant_set_field_strv(sd_json_variant **v, const char *field, char **l);
+
+sd_json_variant *sd_json_variant_find(sd_json_variant *haystack, sd_json_variant *needle);
+
+int sd_json_variant_append_array(sd_json_variant **v, sd_json_variant *element);
+int sd_json_variant_append_arrayb(sd_json_variant **v, ...);
+int sd_json_variant_append_array_nodup(sd_json_variant **v, sd_json_variant *element);
+
+int sd_json_variant_merge_object(sd_json_variant **v, sd_json_variant *m);
+int sd_json_variant_merge_objectb(sd_json_variant **v, ...);
+
+int sd_json_variant_sort(sd_json_variant **v);
+int sd_json_variant_normalize(sd_json_variant **v);
+
+__extension__ typedef enum sd_json_parse_flags_t {
+        SD_JSON_PARSE_SENSITIVE = 1 << 0, /* mark variant as "sensitive", i.e. something containing secret key material or such */
+        _SD_ENUM_FORCE_S64(JSON_PARSE_FLAGS)
+} sd_json_parse_flags_t;
+
+int sd_json_parse_with_source(const char *string, const char *source, sd_json_parse_flags_t flags, sd_json_variant **ret, unsigned *reterr_line, unsigned *reterr_column);
```

Comment:
Maybe better to update our coding style document that `reterr_xyz` is filled on failure.

Reply:
→ https://github.com/systemd/systemd/pull/33296